aospy/automate.py

```diff
@@ -0,0 +1,220 @@
+"""Functionality for specifying and cycling through multiple calculations."""
+from __future__ import print_function
+
+import itertools
+import logging
+
+from .calc import Calc, CalcInterface
+
+try:
+    import colorama
+except ImportError:
+    pass
+try:
+    import multiprocess
+except ImportError:
+    pass
+
+
+def _get_obj_default_or_all(name, parent, attr_name, defaults_attr_name):
+    if name == 'all':
+        return getattr(parent, attr_name).values()
+    elif name == 'default':
+        return getattr(parent, defaults_attr_name).values()
+    else:
+        return getattr(parent, attr_name)[name]
```

Comment:
On that point, the main script and example object library from these commits should work; try running the main script (from inside the example directory so that you don't have to install the object library).